File: src/main/java/MoveZeroes/CodeLove/MoveZeroes.java
package MoveZeroes.CodeLove;

public class MoveZeroes {
    public void moveZeros(int[] nums){
        for (int i = 0, cur = 0; i< nums.length; i++) {
            if (nums[i] !=0) {
                int swap = nums[i];
                nums[i] = nums[cur];
                nums[cur++] = swap;
            }
        }
    }
}

File: src/main/java/MoveZeroes/BBIGBOSS/MoveZeroes.cpp
//
// Created by BBIGBOSS on 2019-06-15.
//

class Solution
{
    public:

    void moveZeroes(vector<int>& nums)
    {
        for (int i = 0, j = 0; j < nums.size(); j++)
            if (nums[j] != 0)
                swap(nums[j], nums[i++]);
    }
};